components/menu/index.css

```diff
@@ -10,170 +10,637 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+  }
+  
+  .spectrum-Menu-itemDescription {
+    color: var(--highcontrast-menu-item-description-color-disabled, var(--mod-menu-item-description-color-disabled, var(--spectrum-menu-item-description-color-disabled)));
+  }
+  
   .spectrum-Menu-itemIcon {
-    /* Don't get smaller, you're an icon! */
-    flex-shrink: 0;
-    align-self: flex-start;
+    fill: var(--highcontrast-menu-item-label-icon-color-disabled, var(--mod-menu-item-label-icon-color-disabled, var(--spectrum-menu-item-label-icon-color-disabled)));
   }
 
-  .spectrum-Icon + .spectrum-Menu-itemLabel,
-  .spectrum-Menu-itemIcon + .spectrum-Menu-itemLabel {
-    margin-inline-start: var(--spectrum-listitem-texticon-icon-gap);
+  &:hover {
+    cursor: default;
 
-    inline-size: calc(
-      100% - var(--spectrum-listitem-texticon-ui-icon-width) -
-        var(--spectrum-listitem-texticon-icon-gap) -
-        var(--spectrum-listitem-textthumbnail-padding-left) -
-        var(--spectrum-alias-workflow-icon-size-m)
-    );
+    .spectrum-Menu-itemIcon {
+      fill: var(--highcontrast-menu-item-label-icon-color-disabled, var(--mod-menu-item-label-icon-color-disabled, var(--spectrum-menu-item-label-icon-color-disabled)));
+    }
   }
 }
 
+.spectrum-Menu-itemIcon {
+  grid-area: iconArea;
+  align-self: center;
+}
+
+.spectrum-Menu-checkmark {
+  grid-area: checkmarkArea;
+}
+
+.spectrum-menu-itemSelection {
+  grid-area: selectedArea;
+}
+
 .spectrum-Menu-itemLabel {
-  flex: 1 1 auto;
-  line-height: var(--spectrum-listitem-texticon-label-line-height);
+  grid-area: labelArea;
+  font-size: var(--mod-menu-item-label-font-size, var(--spectrum-menu-item-label-font-size));
+  color: var(--highcontrast-menu-item-label-content-color-default, var(--mod-menu-item-label-content-color-default, var(--spectrum-menu-item-label-content-color-default)));
+}
+
+.spectrum-Menu-itemValue {
+  grid-area: valueArea;
+}
+
+.spectrum-Menu-itemActions {
+  grid-area: actionsArea;
+}
+
+.spectrum-Menu-chevron {
+  grid-area: chevronArea;
+  align-self: center;
+  block-size: var(--spectrum-menu-item-checkmark-height);
+  height: var(--spectrum-menu-item-checkmark-height);
+  inline-size: var(--spectrum-menu-item-checkmark-width);
+  width: var(--spectrum-menu-item-checkmark-width);
+}
+
+.spectrum-Menu-item--collapsible .spectrum-Menu-chevron {
+  grid-area: chevronAreaCollapsible;
+}
+
+.spectrum-Menu-sectionHeading {
+  grid-area: sectionHeadingArea;
+  grid-column: 1 / -1;
+}
+
+.spectrum-Menu-item--collapsible .spectrum-Menu-sectionHeading {
+  /*
+    start at 5th column, to account for icons & match sectionHeadingArea
+    ". chevronAreaCollapsible . iconArea sectionHeadingArea  .  .  ."
+    stretch to last column
+  */
+  grid-column: 5 / -1;
+}
+
+.spectrum-Menu li:not(.spectrum-Menu-item, .spectrum-Menu-divider) > .spectrum-Menu-sectionHeading {
```

Comment:
Are there times when `.spectrum-Menu-sectionHeading` shouldn't have this padding?